apps/bridge-dapp/src/hooks/useMaxFeeInfo.ts

```diff
@@ -175,26 +173,30 @@ export const useMaxFeeInfo = (
         // On Substrate, we use partial fee dirrectly
         setFeeInfo(gasLimit[currentTypedChain]);
         setIsLoading(false);
-        return;
-      } else if (provider instanceof Web3Provider) {
+      } else if (activeApi instanceof WebbWeb3Provider) {
         const gasAmount = gasLimit[currentTypedChain];
-        const etherProvider = provider.intoEthersProvider();
+        const walletClient = activeApi.walletClient;
+        const publicClient = activeApi.publicClient;
+
+        const { maxFeePerGas, gasPrice, maxPriorityFeePerGas } =
+          await fetchFeeData({
+            chainId: walletClient.chain?.id,
```

Comment:
what if chain.id is undefined?

Reply:
Can you please check for this case.

Reply:
The library manages this, with `chainId` being optional. Also, `fetchFeeData` is a `viem` function. @elkremroot